### PR TITLE
[Resource] Do not check for permission by default

### PR DESCRIPTION
**Pull request by inssein**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | yes
| Related tickets | fixes https://github.com/Sylius/Sylius/issues/5261
| License         | MIT

### CODE REVIEW

**Comment by inssein** [comment]
@pjedrzejewski I just realized that I missed the routes in WebBundle/backend. It seems like the whole WebBundle is going away before the next release, right?

Is there an updated roadmap anywhere?

File: src/Sylius/Bundle/AdminBundle/Resources/config/routing/security.yml [hunk 1]
```diff
@@ -4,6 +4,8 @@ sylius_admin_login:
         _controller: sylius.controller.security:loginAction
         _sylius:
             template: SyliusAdminBundle:Security:login.html.twig
+            permission: true
+
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/AdminBundle/Resources/config/routing/security.yml]: Redundant line

File: src/Sylius/Bundle/AdminBundle/Resources/config/routing/taxon.yml [hunk 2]
```diff
@@ -28,4 +31,6 @@ sylius_admin_taxon_tree:
             template: SyliusAdminBundle:Taxon:index.html.twig
             repository:
                 method: findRootNodes
+            permission: true
+
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/AdminBundle/Resources/config/routing/taxon.yml]: This line should not be added.

File: src/Sylius/Bundle/AdminBundle/Resources/config/routing/zone.yml [hunk 3]
```diff
@@ -35,5 +38,7 @@ sylius_admin_zone_create:
                 route:
                     parameters:
                         type: $type
+            permission: true
+
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/AdminBundle/Resources/config/routing/zone.yml]: Same as above.

File: src/Sylius/Bundle/ResourceBundle/Controller/RequestConfiguration.php [hunk 4]
```diff
@@ -470,7 +470,7 @@ public function getEvent()
     public function hasPermission()
     {
         if (!$this->parameters->has('permission')) {
-            return true;
+            return false;
         }
 
         return false !== $this->parameters->get('permission');
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/ResourceBundle/Controller/RequestConfiguration.php]: `return true === $this->parameters->get('permission');` would be safer approach (e.g. `false !== 'false'` - will return true...).

Comment (michalmarcinkowski) [inline src/Sylius/Bundle/ResourceBundle/Controller/RequestConfiguration.php]: The method `hasPermission` is confusing and actually it checks if permission is disabled. Maybe a better name for the method would be `isPermissionCheckDisabled`? /cc @pjedrzejewski

Comment (michalmarcinkowski) [inline src/Sylius/Bundle/ResourceBundle/Controller/RequestConfiguration.php]: The method `hasPermission` is now used in a wrong context. See [RbacAuthorizationChecker](https://github.com/Sylius/Sylius/blob/147d0169768a91c08979cc6841cf5c1b7e9b5746/src/Sylius/Bundle/RbacBundle/Authorization/RbacAuthorizationChecker.php#L44). Taking into account current logic:

| $this->parameters->has('permission') | $this->parameters->get('permission')  | hasPermission |
| ------------- |------------- |------------
| false      | - | false |
| true      | false | false |
| true      | 'permission' | true |

When I do not set permission

Comment (lchrusciel) [inline src/Sylius/Bundle/ResourceBundle/Controller/RequestConfiguration.php]: Why can't we define some default value as a second parameter? Won't it be equal?
```php
if (!$this->parameters->has('permission')) {
    return false;
}
 
return false !== $this->parameters->get('permission');
```
/\
||
\/
```php
return false !== $this->parameters->get('permission', false);
```

**Comment by inssein** [comment]
@michalmarcinkowski @lchrusciel thanks for the code review - made the changes now.

**Comment by inssein** [comment]
@michalmarcinkowski I only have two tests failing - I can't figure out which config that is using. Any pointers?

**Comment by pjedrzejewski** [comment]
These are RBAC scenarios failing for old admin, because after this change all routes in WebBundle do not have any permission set... So we either set it to true on all routes or wait a bit with this merge until SyliusWebBundle is removed.

**Comment by inssein** [comment]
@pjedrzejewski alright if I get some more time I will go ahead and add it to the old routes. Is there a roadmap anywhere? I'd love to be able to roughly plan. My sylius store is currently at 0.18, but working with master on a branch.

**Comment by pjedrzejewski** [comment]
Thank you Hussein! 👍

**Pull request merged**